src/backend/joanie/core/models/accounts.py

```diff
@@ -45,3 +46,13 @@ class Meta:
 
     def __str__(self):
         return f"{self.address}, {self.postcode} {self.city}, {self.country}"
+
+    def save(self, *args, **kwargs):
+        # no more one main=True per Owner
+        if self.main and self.owner.addresses.filter(main=True):
+            self.owner.addresses.filter(main=True).update(main=False)
```

Comment:
You should check but I think currently there are two db accesses to get addresses with `main` attribute sets to `True`; You should store `self.owner.addresses.filter(main=True)` in a variable to avoid that.

Reply:
I just checked. I don't have more db access without use a variable to store `self.owner.addresses.filter(main=True)`
But I think it's more easy to read with a variable. I change that.

Reply:
Ok evaluation of queryset is still obscure for me.
Did you see the `exists` method which seems faster to check if a queryset has a result.
https://docs.djangoproject.com/en/3.2/ref/models/querysets/#django.db.models.query.QuerySet.exists